packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,528 @@
+  });
+
+  test('getNestedStack() throws an exception when getting a resource that does not exist in the template', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(() => {
+      parentTemplate.getNestedStack('FakeStack');
+    }).toThrow(/Nested Stack with logical ID 'FakeStack' was not found/);
+  });
+
+  test('getNestedStack() throws an exception when getting a resource that exists in the template, but is not a Stack', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(() => {
+      const childTemplate = parentTemplate.getNestedStack('ChildStack').includedTemplate;
+      childTemplate.getNestedStack('BucketImport');
+    }).toThrow(/Resource with logical ID 'BucketImport' is not a CloudFormation Stack/);
+  });
+
+  test('getNestedStack() throws an exception when getting a resource that exists in the template, but was not specified in the props', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(() => {
+      parentTemplate.getNestedStack('AnotherChildStack').includedTemplate.getNestedStack('BucketImport');
+    }).toThrow(/Nested Stack 'AnotherChildStack' was not included in the nestedStacks property when including the parent template/);
+  });
+
+  test('correctly handles renaming of references across nested stacks', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('cross-stack-refs.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+    const cfnBucket = parentTemplate.getResource('Bucket');
+    cfnBucket.overrideLogicalId('DifferentBucket');
+    const parameter = parentTemplate.getParameter('Param');
+    parameter.overrideLogicalId('DifferentParameter');
+
+    expect(stack).toHaveResourceLike('AWS::CloudFormation::Stack', {
+      "Parameters": {
+        "Param1": {
+          "Ref": "DifferentParameter",
+        },
+        "Param2": {
+          "Fn::GetAtt": ["DifferentBucket", "Arn"],
+        },
+      },
+    });
+  });
+
+  test('ensure that getResource() returns a CfnStack', () => {
```

Comment:
The description here is weird...

```suggestion
  test('returns the CfnStack object from getResource() for a nested stack that was not in the nestedStacks property', () => {
```